core/src/main/java/io/confluent/rest/FileWatcher.java

```diff
@@ -100,9 +100,15 @@ private void handleNextWatchNotification() throws InterruptedException {
         log.debug("Watch event is OVERFLOW");
         continue;
       }
-      WatchEvent<Path> ev = (WatchEvent<Path>)event;
-      Path changed = this.file.getParent().resolve(ev.context());
-      log.info("Watch file change: " + ev.context() + "=>" + changed);
+
+      if (event.context() != null && !(event.context() instanceof Path)) {
+        throw new ClassCastException("Expected `event.context()` to be an instance of `Path`, "
+            + "but it is " + event.context().getClass());
```

Comment:
super-nit: My OCD is tingling that `Path` is not fully qualified unlike the context class name, and that the context class name would not be wrapped in backticks in the exception message, but I'm totally OK with you leaving it as-is.